src/Featurizers/Structs.h

```diff
@@ -240,6 +299,56 @@ bool SingleValueSparseVectorEncoding<T>::operator!=(SingleValueSparseVectorEncod
     return (*this == other) == false;
 }
 
+
+// ----------------------------------------------------------------------
+// |
+// |  CSREncoding
+// |
+// ----------------------------------------------------------------------
+template <typename T>
+CSREncoding<T>::CSREncoding() {
+    RowIndex.push_back(0);

```

Comment:
>RowIndex.push_back(0); [](start = 4, length = 22)

This should be implemented in terms of the functionality below - it should participate in the same validation logic.